src/Widgets/PasswordEditor.vala

```diff
@@ -69,88 +65,108 @@ namespace SwitchboardPlugUserAccounts.Widgets {
             orientation = Gtk.Orientation.VERTICAL;
             row_spacing = 3;
             add (pw_label);
-            add (new_pw_entry);
+            add (pw_entry);
             add (pw_levelbar);
             add (pw_error_revealer);
             add (confirm_label);
             add (confirm_entry);
             add (confirm_entry_revealer);
             add (show_pw_check);
+            show_all ();
 
-            show_pw_check.bind_property ("active", new_pw_entry, "visibility", GLib.BindingFlags.DEFAULT);
+            show_pw_check.bind_property ("active", pw_entry, "visibility", GLib.BindingFlags.DEFAULT);
             show_pw_check.bind_property ("active", confirm_entry, "visibility", GLib.BindingFlags.DEFAULT);
 
-            new_pw_entry.changed.connect (compare_passwords);
-            confirm_entry.changed.connect (compare_passwords);
+            pw_entry.changed.connect (() => {
+                pw_entry.is_valid = check_password ();
+                confirm_entry.is_valid = confirm_password ();
+                validate_form ();
```

Comment:
Perhaps put these two lines in a method like `validate_confirm_entry`, since these are also present in the confirm entry changed lambda?